File: src/test/java/warmke/mike/leet/p8/SolutionTest.java
package warmke.mike.leet.p8;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SolutionTest {

    Solution solution;

    @BeforeEach
    public void setup() {
        solution = new Solution();
    }

    @Test
    public void test() {
        assertEquals(42, solution.myAtoi("42"));
        assertEquals(-42, solution.myAtoi("-42"));
        assertEquals(4193,  solution.myAtoi("4193 with words"));
        assertEquals(Integer.MIN_VALUE,  solution.myAtoi("-91283472332"));
        assertEquals(0, solution.myAtoi("words and 987"));
        assertEquals(0, solution.myAtoi("+-12"));
        assertEquals(3, solution.myAtoi("3.14159"));
        assertEquals(-12, solution.myAtoi("  -0012a42"));
        assertEquals(-2147483647, solution.myAtoi("-2147483647"));
        assertEquals(Integer.MAX_VALUE, solution.myAtoi("20000000000000000000"));
    }

    @Test
    public void stuff() {
        System.out.println(Character.isDigit('+'));
        System.out.println(Character.isDigit('-'));
        System.out.println(Character.isDigit('.'));
        System.out.println(((int)'9') >>> 16);
    }
}

File: src/test/java/warmke/mike/leet/p2131/SolutionTest.kt
package warmke.mike.leet.p2131

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

internal class SolutionTest {

    val s = Solution()

    @Test
    fun test() {
//        assertEquals(6, s.longestPalindrome(arrayOf("lc","cl","gg")))
        assertEquals(4, s.longestPalindrome(arrayOf("bb","bb")))
        assertEquals(8, s.longestPalindrome(arrayOf("ab","ty","yt","lc","cl","ab")))
        assertEquals(2, s.longestPalindrome(arrayOf("cc","ll","xx")))
        assertEquals(22, s.longestPalindrome(arrayOf("dd","aa","bb","dd","aa","dd","bb","dd","aa","cc","bb","cc","dd","cc")))
        assertEquals(26, s.longestPalindrome(arrayOf("ll","lb","bb","bx","xx","lx","xx","lx","ll","xb","bx","lb","bb","lb","bl","bb","bx","xl","lb","xx")))
    }
}